src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -964,7 +995,10 @@ public void ExecuteSyntaxNodeActions<TLanguageKindEnum>(
 
             var diagReporter = GetAddSemanticDiagnostic(model.SyntaxTree, spanForContainingTopmostNodeForAnalysis, analyzer, cancellationToken);
 
-            using var _ = PooledDelegates.GetPooledFunction((d, ct, arg) => arg.self.IsSupportedDiagnostic(arg.analyzer, d, ct), (self: this, analyzer), out Func<Diagnostic, CancellationToken, bool> isSupportedDiagnostic);
+            using var _ = PooledDelegates.GetPooledFunction(
+                static (d, ct, arg) => arg.self.IsSupportedDiagnostic(arg.analyzer, d, ct),
```

Comment:
note: i don't like the use of 'ct' as a name here, as it means code can accidentally capture an outer variable called 'cancellationToken'.  However, as these are all static lambdas now, we don't have to worry about this.  So i kept the name the same.